internal/controller/streams.go

```diff
@@ -136,6 +136,7 @@ func ServerIsHealthy() bool {
 
 // StreamResources handles the resource stream.
 func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger, cancel context.CancelFunc) error {
+	defer cancel()
```

Comment:
This is unnecessary. The caller already does that here:
https://github.com/illumio/cloud-operator/blob/d0075cbd712ad3808e64be909e9fc28040ea6d8c/internal/controller/streams.go#L455